packages/cli/src/config.ts

```diff
@@ -0,0 +1,79 @@
+import { getLogger } from "@logtape/logtape";
+import { join } from "@std/path";
+
+const logger = getLogger(["fedify", "cli", "config"]);
+
+interface HttpConfig {
+  /** http header User-Agent */
+  userAgent?: string;
+  /** http request timeout */
+  timeout?: number;
+  /** auto follow redirects mode */
+  followRedirects?: boolean;
+}
+
+interface FormatConfig {
+  /** default output format */
+  default?: string;
+}
+
+/**
+ * @description config for cli
+ * @example
+ * {
+ *   "http": {
+ *     "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36",
+ *     "timeout": 10000,
+ *     "followRedirects": true
+ *   },
+ *   "format": {
+ *     "default": "json"
+ *   },
+ *   "cacheDir": "./cache",
+ *   "verbose": true
+ * }
+ */
+export interface Config {
+  /** HTTP related configuration */
+  http?: HttpConfig;
+  /** Output format configuration */
+  format?: FormatConfig;
+  /** cache directory */
+  cacheDir?: string;
+  /** verbose mode */
+  verbose?: boolean;
+}
+
+export async function loadConfig(): Promise<Config> {
+  const currentDir = Deno.cwd();
+
+  // for more details, see https://wiki.archlinux.org/title/XDG_Base_Directory
+  const homeDir = Deno.env.get("XDG_CONFIG_HOME") ??
```

Comment:
On Windows, it's better to use `AppData` instead of `XDG_CONFIG_HOME`. See also <https://learn.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables>.